### PR TITLE
Added escaping to printf and sprintf functions

### PR DESCRIPTION
**Pull request by dashaluna**

### CODE REVIEW

File: includes/class-content-toc.php [hunk 1]
```diff
@@ -433,7 +433,7 @@ protected function insert_anchors( $post_content, $toc_items_matches ) {
 			$anchors[] = sprintf(
 				'<a name="heading-%s"%s></a>',
 				esc_attr( $key_current ),
-				$this->tag_class( $this->settings['anchor_class'] ),
+				esc_attr( $this->tag_class( $this->settings['anchor_class'] ) ),
 				$match_set[0]
```
Comment (dashaluna) [inline includes/class-content-toc.php]: @johnbillion do I need to use `wp_kses_post` on `$match_set[0]`? It's the original header in the post content that I matched, not sure if I need to escape it in this case?

Comment (johnbillion) [inline includes/class-content-toc.php]: This parameter isn't used so can be removed.

Comment (dashaluna) [inline includes/class-content-toc.php]: Cleaned up!

File: includes/class-content-toc.php [hunk 2]
```diff
@@ -233,8 +233,8 @@ public function filter_content( $post_content, $shortcode_atts ) {
 			$list_html = sprintf(
 				'<%1$s%2$s>%3$s</%1$s>',
 				esc_attr( $this->settings['list_tag'] ),
-				$this->tag_class( $this->settings['list_class'] ),
-				$items_html
+				esc_attr( $this->tag_class( $this->settings['list_class'] ) ),
```
Comment (johnbillion) [inline includes/class-content-toc.php]: The `tag_class()` method returns a complete class attribute (eg. `class="foo"`), so passing it through `esc_attr()` means it won't work and will end up as `class=&quot;foo&quot;`. Same applies to the changes below too.

Comment (johnbillion) [inline includes/class-content-toc.php]: I would alter `tag_class()` so it just returns the attribute value, and then use `class="%2$s"` in your formatted strings.

Comment (dashaluna) [inline includes/class-content-toc.php]: I was trying to avoid situations with empty classes, so my HTML would be clear from `class=""`

**Comment by dashaluna** [comment]
@johnbillion all the PRs are ready for the review & merge ;)

**Pull request merged**